sweep/sweeper.go

```diff
@@ -547,8 +574,65 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 
 			bestHeight = epoch.Height
 
-			log.Debugf("New block: height=%v, sha=%v",
-				epoch.Height, epoch.Hash)
+			log.Debugf("New block: height=%v, sha=%v", epoch.Height,
+				epoch.Hash)
+
+			// In order to correctly meet the confirmation targets
+			// of inputs, we'll bump their fee rates accordingly.
+			//
+			// NOTE: Meeting the confirmation target is a bit
+			// inaccurate at the moment due to the sweeper randomly
+			// rescheduling publish attempts for inputs.
+			for input, confTarget := range s.confTargetInputs {
+				pendInput := s.pendingInputs[input]
+
+				// We'll only proceed to query for a new fee
+				// rate if we don't exceed the threshold of
+				// confirmations left in order to do so.
+				expectedConfHeight := pendInput.initialHeight +
```

Comment:
I think we need to reset `initialHeight` here. Otherwise, after the first retarget period, we'll continually retarget it each block.

Reply:
Done.